qvisor/src/runc/runtime/signal_handle.rs

```diff
@@ -115,18 +119,40 @@ extern "C" fn handle_sigintAct(
             panic!("get signal 11");
         }
 
-        let signal = SignalArgs {
-            Signo: signal,
-            CID: ROOT_CONTAINER_ID.lock().clone(),
-            PID: 0,
-            Mode: if console {
-                SignalDeliveryMode::DeliverToForegroundProcessGroup
-            } else {
-                SignalDeliveryMode::DeliverToProcess
-            },
-        };
-
-        VMS.lock().Signal(signal);
+        #[cfg(not(feature = "cc"))]{
+            let signal = SignalArgs {
+                Signo: signal,
+                CID: ROOT_CONTAINER_ID.lock().clone(),
+                PID: 0,
+                Mode: if console {
+                    SignalDeliveryMode::DeliverToForegroundProcessGroup
+                } else {
+                    SignalDeliveryMode::DeliverToProcess
+                },
+            };
+
+            VMS.lock().Signal(signal);
+        }
+
+        #[cfg(feature = "cc")]
+        if is_cc_enabled(){
+            let res = SignalProcess(&ROOT_CONTAINER_ID.lock().clone(), 0, signal, console);
```

Comment:
This could be optimized to reduce code repetition.

Reply:
Sure, I can extract a function here.